dictionary/dictionary.go

```
@@ -194,12 +193,12 @@ func (d *Dictionary) Lookup(key, schemaVersion, extVersion int) (*ContainerDefin
 		result, err := d.search(key, schemaVersion, extVersion)
 		if err == nil {
 			d.Add(result)
-			return result, true
+			return result, true, nil
 		}
-		log.Errorf("Fail to lookup %v, %v", ckey, err)
+		return nil, found, err
```

The new `error` return seems unnecessary. Didn't the bool `found` already indicate the same thing?

---

the error might cause by invalid data type, if we didn't return error for logging, would it actual error get hidden?

---

I want to print error message look like this that have a bit more detail. 
```
ERRO[2024-07-19T06:20:27Z] [10.244.1.7:14061->10.244.1.1:17320] Error lookup definition key 447 schemaVersion 5263 extVersion 1: Error field SubscriberId: Unknown datatype: object id                                                                        
INFO[2024-07-19T06:20:57Z] [10.244.1.7:14061->10.244.1.1:17320] Connection closed
```
maybe I can remove the bool and replace with error

---

ok make sense to replace `found` bool with `err`